matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/network/Request.kt

```diff
@@ -73,14 +71,15 @@ internal suspend inline fun <DATA> executeRequest(globalErrorReceiver: GlobalErr
                     ?.also { unrecognizedCertificateException -> throw unrecognizedCertificateException }
 
             currentRetryCount++
-
-            if (exception is Failure.ServerError &&
-                    exception.httpCode == 429 &&
-                    exception.error.code == MatrixError.M_LIMIT_EXCEEDED &&
-                    currentRetryCount < maxRetriesCount) {
-                // 429, we can retry
-                delay(exception.getRetryDelay(1_000))
-            } else if (canRetry && currentRetryCount < maxRetriesCount && exception.shouldBeRetried()) {
+            // Tchap: We don't add a delay in Tchap
+//            if (exception is Failure.ServerError &&
+//                    exception.httpCode == 429 &&
+//                    exception.error.code == MatrixError.M_LIMIT_EXCEEDED &&
+//                    currentRetryCount < maxRetriesCount) {
+//                // 429, we can retry
+//                   delay(exception.getRetryDelay(1_000))
+//            } else
+            if (canRetry && currentRetryCount < maxRetriesCount && exception.shouldBeRetried()) {
```

Comment:
I disagree about this change.
If I understood correctly, you commented this block about `exception.error.code == MatrixError.M_LIMIT_EXCEEDED`, in order to prevent the application to be blocked until the end of the delay provided by the exception (which is 30 mins in our case). If this is true, this means `canRetry` and/or `exception.shouldBeRetried()` was false, which let you trigger the exception.

I think there is a bug in Element-android sdk here, we should check `exception.shouldBeRetried()` and `canRetry` booleans even when `exception.error.code == MatrixError.M_LIMIT_EXCEEDED` 

About my suggestion:
- we have to check whether the login request is triggered by disabling the retry option
- we have to test the other cases where `M_LIMIT_EXCEEDED` error may be received to be sure we don't disturb them. For example when you send several messages in a room in a very short delay, you will trigger this error. I think this request (send message) enables the retry

Reply:
For more context the new retry policy regarding 429 has been added https://github.com/vector-im/element-android/pull/3142.

Reply:
And in case of 429, the SDK automatically retry, whatever the value of `canRetry`